util/src/main/java/io/kubernetes/client/util/credentials/TokenFileAuthentication.java

```diff
@@ -0,0 +1,59 @@
+package io.kubernetes.client.util.credentials;
+
+import io.kubernetes.client.openapi.ApiClient;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class TokenFileAuthentication implements Authentication, Interceptor {
+  private String file;
+  private String token;
+  private Instant expiry;
+
+  public TokenFileAuthentication(String file) {
+    this.expiry = Instant.MIN;
+    this.file = file;
+  }
+
+  private String getToken() {
+    if (Instant.now().isAfter(this.expiry)) {
+      try {
+        this.token =
+            new String(Files.readAllBytes(Paths.get(this.file)), Charset.defaultCharset()).trim();
+        expiry = Instant.now().plusSeconds(60);
+      } catch (IOException ie) {
+        throw new RuntimeException("Cannot read file: " + this.file);
+      }
+    }
+
+    return this.token;
+  }
+
+  public void setExpiry(Instant expiry) {
+    this.expiry = expiry;
+  }
+
+  public void setFile(String file) {
+    this.file = file;
+  }
+
+  @Override
+  public void provide(ApiClient client) {
+    OkHttpClient withInterceptor = client.getHttpClient().newBuilder().addInterceptor(this).build();
```

Comment:
the # of interceptors seems to be growing unboundly upon every `provide` call, isn't it?

Reply:
yes, the assumption is made that the `provide` function is called only once for each in-cluster client. i am not sure when a user will call this twice on the same client instance.

Reply:
>  the assumption is made that the provide function is called only once for each in-cluster client.

yea, that's true